ci: prevent concurrent version bumps, and await on release before we …

joshuef opened the pull request:
…complete the bump

The git commit is pushed earlier, and should trigger a release run which we then check for

## Description

<!-- reviewpad:summarize:start -->
### Summary generated by Reviewpad on 19 Oct 23 13:43 UTC
This pull request updates the version bump workflow to prevent concurrent version bumps and adds a wait step for the release workflow before completing the bump. This ensures that the release run is triggered by the earlier commit and is checked before proceeding.
<!-- reviewpad:summarize:end -->

Yoga07 commented on sn/src/client/utils/test_utils/mod.rs:
```diff
@@ -44,6 +45,25 @@ where
     }
 }
 
+fn retry<R, E, Fn, Fut>(
```
Just curious as to why this was moved to another module?

bochaco replied:
because it's only used within that module now that we don't use it from `send_query_with_retry_count`

bochaco replied:
...and we may not even needed in that other module either, if we are happy with the logic/pattern of this PR

bochaco commented on sn/src/client/client_api/queries.rs:
```diff
@@ -52,55 +50,41 @@ impl Client {
-            "Setting up query retry, initial interval is: {:?}",
-            starting_query_timeout
-        );
+        let attempt_timeout = self.query_timeout.div_f32(MAX_RETRY_COUNT + 1.0);
```
it was just to be strict on the "retry" meaning, first attempt wouldn't be a retry

joshuef committed:
ci: prevent concurrent version bumps, and await on release before we complete the bump

The git commit is pushed earlier, and should trigger a release run which we then check for
